src/app/views/canvas/mark_editor.tsx

```diff
@@ -313,6 +320,24 @@ export class SingleMarkView
     });
   }
 
+  public doCustomZoom(cx: number, cy: number, width: number, height: number) {
+    const newCX = this.props.width / 2 - cx;
+    const newCY = this.props.height / 2 + cy;
+    const newScale =
+      this.props.width > this.props.height
+        ? this.props.height / height
+        : this.props.width / width;
+
+    this.setState({
```

Comment:
See similar comment above on `setState` being called here and then in `doZoom`.

Reply:
Fixed